File: src/filedata.py
import os

class filedata:
    def __init__(self, data: dict):
        self.data = data
        self.__filename = "inventario.txt"
        self.__header = ("Hostname", "IP", "MACaddress", "RAM", "Procesador", 
                         "Sistema", "Responsable", "Cubículo", "Inventario","Desktop")
        self.__new_file = not os.path.exists(self.__filename)

    def export_txt(self) -> None:
        with open(self.__filename, "a") as f:
            if self.__new_file:
                f.write("*".join(self.__header) + "\n")
                f.write("-" * 70 + "\n")  
                self.__new_file = False
            row = [self.data.get(h, "N/A") for h in self.__header]
            f.write("*".join(row) + "\n")

File: src/filedata.c
#include "filedata.h"
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

void export_txt(SystemInfo *info) {
    FILE *f;
    bool new_file = false;

   
    f = fopen("inventario.txt", "r");
    if (!f) {
        new_file = true; 
    } else {
        fclose(f);
    }

    f = fopen("inventario.txt", "a");
    if (!f) return;

    if (new_file) {
        fprintf(f, "Hostname*IP*MACaddress*RAM*Procesador*Sistema*Responsable*Cubículo*Inventario*Desktop\n");
    }

    const char *ip = (strlen(info->ip) > 0) ? info->ip : "";
    const char *mac = (strlen(info->mac) > 0) ? info->mac : "";

    fprintf(f, "%s*%s*%s*%s*%s*%s*%s*%s*%s*%s\n",
        info->hostname,
        ip,
        mac,
        info->ram,
        info->cpu,
        info->sistema,
        info->responsable,
        info->cubiculo,
        info->inventario,
        info->desktop
    );

    fclose(f);
}
